src/components/Login/Login.tsx

```diff
@@ -18,7 +18,8 @@ const Login = () => {
 	const handleSubmit = (e: any) => {
 		e.preventDefault();
 		
-		if (validate()) {
+		validate()
+		if (!validateLoginData(user).isError) {
```

Comment:
There was a bug that required me to have to double-click submit, because it wouldn't update the isError field through validate method quick enough before being checked here in the if statement. So I called the validation method directly to have the right value for the isError check

Reply:
The way to fix this would be to validate fields when the user stops typing or once they use focus, not duplicating the same code. The reason why this happens is because setting the state is an async operation, but once it finishes, all data will be rerendered by default. Also, no need to call the validateLoginData(user) function so many times. You can call it just once, save that result into a variable and then distribute the attributes wherever you need to from there.